Kernel/Net/IPv4Socket.cpp

```diff
@@ -631,7 +635,7 @@ ErrorOr<void> IPv4Socket::ioctl(OpenFileDescription&, unsigned request, Userspac
                 return EAFNOSUPPORT;
             if ((route.rt_flags & (RTF_UP | RTF_GATEWAY)) != (RTF_UP | RTF_GATEWAY))
                 return EINVAL; // FIXME: Find the correct value to return
-            adapter->set_ipv4_gateway(IPv4Address(((sockaddr_in&)route.rt_gateway).sin_addr.s_addr));
+            update_routing_table(destination, gateway, genmask, adapter, UpdateTable::Set);
```

Comment:
Shouldn't we be returning error if update the routing table fails (out of memory for example?)

Reply:
I think I got this one right?